app/models/spree/calculator/taxjar_calculator.rb

```diff
@@ -94,9 +94,9 @@ def cache_response(taxjar_response, order, address, item = nil)
 
       def cache_key(order, item, address)
         if item.is_a?(Spree::LineItem)
-          ['Spree::LineItem', order.id, item.id, address.state.id, address.zipcode, item.amount, :amount_to_collect]
+          ['Spree::LineItem', order.id, item.id, address.state.id, address.zipcode, item.taxable_amount, :amount_to_collect]
         else
-          ['Spree::Shipment', order.id, item.id, address.state.id, address.zipcode, item.cost, :amount_to_collect]
+          ['Spree::Shipment', order.id, item.id, address.state.id, address.zipcode, item.cost, item.adjustments.where.not(source_type: "Spree::TaxRate").pluck(:amount).sum.to_f, :amount_to_collect]
```

Comment:
@himanshumishra31 Please avoid firing a query here and use already loaded objects to find the sum